tiledb/sm/c_api/tiledb_enum.h

```diff
@@ -36,6 +36,12 @@
     /** Write query */
     TILEDB_QUERY_TYPE_ENUM(WRITE) = 1,
     /** Delete query */
+    #if (_MSC_VER)
+    // note: 'DELETE' is #define'd somewhere within windows headers as
+    // something resolving to '(0x00010000L)', which causes problems with
+    // query_type.h which does not qualify the 'id' like tiledb.h does.
+    #undef DELETE
+    #endif
```

Comment:
This test is absolutely going to fail if we're compiling on Windows with a compiler other than MSVC.

I had to deal with this same problem on a current working branch of mine. Instead of patching it at the point of the symptom as is done here, I put a wall around it at the point of origin in `tdb_catch.h`. Before inclusion of the Catch header, I detected whether `DELETE` was defined and saved its value. After inclusion, I restored whatever the definition state was. This will work in arbitrary situation, not just the one for query types.

Reply:
MSVC specific test being changed.

I have now seen your approach, but I think it is still prone to failure.  Even in your approach, in one path you undef what would have presumably been from windows.h.  And the path in which you preserve something, you may or not be preserving the definition being touched here.
When code reaches here, if DELETE is defined it needs to be undefined, as compilation shouldn't be here twice due to compilation guards in tiledb.h which includes tiledb_enum.h, and any previous definition should therefore be some incorrect definition.

Reply:
The reason I suppressed the symbol at the point of origin was to avoid masking a defect if that same symbol becomes visible in the future because some (future) dependency were also to bleed Windows header symbols. If that were ever to be the case, then we wouldn't know about it until some other problem with it cropped up.

I quite like the approach you have here, which is to detect the problem before the compiler does and to provide a specific and informative error message. It doesn't mask a problem but does prevents one from sowing unnecessary confusion.